app/controllers/application_controller.rb

```diff
@@ -2,4 +2,13 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  before_action :authenticate_user!
+
+  google_signin = "pages/googleallow.html.erb"
+
+  def after_sign_up_path_for(google_signin)
```

Comment:
For devise, i think the proper method is: 

```
after_sign_in_path_for(resource)
```

That expects a user object. You set google_signin to be a path. When you call on the next line:

```
user_path(google_signin)
```

rails doesn't know how to interpret that. user_path NEEDS an id.